### PR TITLE
[WIP] Use `const` for requires

### PR DESCRIPTION
**Pull request by jlord**
We also inlined a few vars that coffee had listed at the top of each file. 

A part of #4065

### CODE REVIEW

File: atom/browser/api/lib/app.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+const Menu = require('electron').Menu;
+const EventEmitter = require('events').EventEmitter;
+
+const bindings = process.atomBinding('app');
+const downloadItemBindings = process.atomBinding('download_item');
+const app = bindings.app;
+
+var slice = [].slice;
+
+var appPath;
```
Comment (kevinsawicki) [inline atom/browser/api/lib/app.js]: I think this was the first file we worked on, do we want to inline these now like we ended up doing in the other files?

Comment (jlord) [inline atom/browser/api/lib/app.js]: oh yeah, I'll do that :+1:

File: atom/browser/api/lib/app.js [hunk 2]
```diff
@@ -54,16 +52,17 @@ app.getAppPath = function() {
...
   return app.on(name, function() {
     var args, event, webContents;
     event = arguments[0], webContents = arguments[1], args = 3 <= arguments.length ? slice.call(arguments, 2) : [];
     return webContents.emit.apply(webContents, [name, event].concat(slice.call(args)));
   });
 };
+var i, len;
 for (i = 0, len = ref1.length; i < len; i++) {
-  name = ref1[i];
+  var name = ref1[i];
```
Comment (etiktin) [inline atom/browser/api/lib/app.js]: The var deceleration should be outside of the loop (line 63).

File: atom/renderer/api/lib/desktop-capturer.js [hunk 3]
```diff
@@ -1,16 +1,15 @@
-var getNextId, ipcRenderer, isValid, nativeImage, nextId, ref,
-  indexOf = [].indexOf || function(item) { for (var i = 0, l = this.length; i < l; i++) { if (i in this && this[i] === item) return i; } return -1; };
+const ipcRenderer = require('electron').ipcRenderer;
+const nativeImage = require('electron').ref.nativeImage;
```
Comment (kevinsawicki) [inline atom/renderer/api/lib/desktop-capturer.js]: Looks like there is a `.ref` still in here left over from before.

File: atom/browser/api/lib/dialog.js [hunk 4]
```diff
@@ -164,8 +162,9 @@ module.exports = {
 };
 
 // Mark standard asynchronous functions.
-ref1 = ['showMessageBox', 'showOpenDialog', 'showSaveDialog'];
+var ref1 = ['showMessageBox', 'showOpenDialog', 'showSaveDialog'];
+var j, len
 for (j = 0, len = ref1.length; j < len; j++) {
-  api = ref1[j];
+  var api = ref1[j];
```
Comment (etiktin) [inline atom/browser/api/lib/dialog.js]: The var should be declared outside of the loop.

File: atom/renderer/api/lib/desktop-capturer.js (hunk 3 above)
Comment (jlord) [inline atom/renderer/api/lib/desktop-capturer.js]: :scream:

File: atom/browser/api/lib/dialog.js (hunk 4 above)
Comment (jlord) [inline atom/browser/api/lib/dialog.js]: Thanks for catching these, I'll go back through and look more closely for these cases. 

We're taking the clean up bit by bit and will get to loops too :+1:

File: atom/browser/api/lib/dialog.js [hunk 5]
```diff
@@ -164,8 +162,9 @@ module.exports = {
 };
 
 // Mark standard asynchronous functions.
-ref1 = ['showMessageBox', 'showOpenDialog', 'showSaveDialog'];
+var ref1 = ['showMessageBox', 'showOpenDialog', 'showSaveDialog'];
+var j, len
```
Comment (kevinsawicki) [inline atom/browser/api/lib/dialog.js]: Super minor, but missing a `;` here

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**